text/3716-target-modifiers.md

```diff
@@ -0,0 +1,517 @@
+- Feature Name: `target_modifiers`
+- Start Date: 2024-10-24
+- RFC PR: [rust-lang/rfcs#3716](https://github.com/rust-lang/rfcs/pull/3716)
+- Rust Issue: None
+
+# Summary
+[summary]: #summary
+
+* We introduce the concept of "target modifier". A target modifier is a flag
+  where it may be unsound if you link together two compilation units that
+  disagree on the flag.
+* We fail the build if rustc can see two Rust compilation units that do not
+  agree on the exact set of target modifier flags.
+* There are already several existing flags that could fall into this category.
+  There are also hypothetical new flags that do.
+* The error can be silenced using the `-Cunsafe-allow-abi-mismatch` escape
```

Comment:
Maybe a name like `-Cunsafe-allow-modifier-mismatch` would be better since it is more general? Some of the included ideas don't necessarily seem to fall under ABI.

Reply:
Yeah, maybe. But I've tried to avoid such naming as explained in [the teaching section](https://github.com/Darksonn/rfcs/blob/target-modifiers/text/3716-target-modifiers.md#teaching).